tests/unit/test_main.py

```diff
@@ -713,3 +713,67 @@ def test_traces_no_action(self, mock_stored_flows):
         assert len(result) == 2
         assert result[0][-1]['type'] == "incomplete"
         assert result[1][-1]['type'] == "incomplete"
+
+    @patch("napps.amlight.sdntrace_cp.main.get_stored_flows")
+    def test_get_traces_str_vlan(self, mock_stored_flows):
+        """Test traces rest call."""
+        api = get_test_client(get_controller_mock(), self.napp)
+        url = f"{self.server_name_url}/traces/"
+
+        payload = [{
+            "trace": {
+                "switch": {
+                    "dpid": "00:00:00:00:00:00:00:01",
+                    "in_port": 1
+                    },
+                "eth": {"dl_vlan": "untagged"},
```

Comment:
I'd expect that this test case would fail if you remove the "eth" layer representing an untagged packet too, could you fix this? It's related to my prior comment

Reply:
Since we're not changing the API we shouldn't be accepting `"dl_vlan": "untagged"` in the request payload body. Could you remove this from the test case? We haven't added complete validation yet, but this when validation is in place this is supposed to fail too.